### PR TITLE
RHBRMS-2476: Allow users to edit project's GAV information in business central Project Editor when it has a parent

### PR DESCRIPTION
**Pull request by karreiro**
This PR adds a new system property that allows users to edit GAV information when it has a parent GAV.

### CODE REVIEW

File: guvnor-project/guvnor-project-api/src/main/java/org/guvnor/common/services/project/service/ProjectRepositoryResolver.java [hunk 1]
```diff
@@ -31,6 +31,8 @@
 
     String CONFLICTING_GAV_CHECK_DISABLED = "org.guvnor.project.gav.check.disabled";
 
+    String CHILD_GAV_EDITION_ENABLED = "org.guvnor.project.gav.child.edit.enabled";
```
Comment (manstis) [inline guvnor-project/guvnor-project-api/src/main/java/org/guvnor/common/services/project/service/ProjectRepositoryResolver.java]: ```CHILD_GAV_EDIT_ENABLED``` would be more in keeping with the System Property and function... but it is me being a bit picky....

File: guvnor-project/guvnor-project-client/src/main/java/org/guvnor/common/services/project/client/POMEditorPanel.java [hunk 2]
```diff
@@ -58,18 +59,21 @@ public void setPOM( POM model,
...
+        view.enableArtifactID();
+        view.enableVersion();
+
         if ( model.hasParent() ) {
             view.setParentGAV( model.getParent() );
             view.showParentGAV();
-            view.disableGroupID( "" );
-            view.enableArtifactID();
-            view.disableVersion( "" );
+            if ( !isChildGAVEditionEnabled() ) {
```
Comment (manstis) [inline guvnor-project/guvnor-project-client/src/main/java/org/guvnor/common/services/project/client/POMEditorPanel.java]: Your use of the word "edition" is wrong (but then again, don't even ask me what it is is Portuguese!) Personally I'd replace all occurrences of "edition" with "edit" e.g. ```isChildGAVEditEnabled()``` etc

File: guvnor-project/guvnor-project-client/src/main/java/org/guvnor/common/services/project/client/ProjectEntryPoint.java [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+        this.appConfigService = appConfigService;
+    }
+
+    @AfterInitialization
+    public void startApp() {
+        loadPreferences();
+    }
+
+    private void loadPreferences() {
+        appConfigService.call( new RemoteCallback<Map<String, String>>() {
```
Comment (manstis) [inline guvnor-project/guvnor-project-client/src/main/java/org/guvnor/common/services/project/client/ProjectEntryPoint.java]: As an aside, you could use lamdas here instead..

**Comment by manstis** [comment]
Nice test coverage.. just a minor comment about "edit" vs "edition"... would you mind changing please?

**Comment by karreiro** [comment]
jenkins retest this

**Comment by karreiro** [comment]
jenkins retest this

**Comment by karreiro** [comment]
Hi @manstis,

I've already done the minor fix about "edit" vs "edition". However, when I pushed the commit, the build started becoming unstable.

Every tests are passing localy, but two of them are not passing at the CI. :(

For while I' removed the following assertions (that are working localy):

```
// guvnor-project/guvnor-project-client/src/test/java/org/guvnor/common/services/project/client/POMEditorPanelTest.java

@@ -102,6 +102,9 @@ public class POMEditorPanelTest {
         verify( view ).setGAV( gavModel.getGav() );
         verify( view ).setTitleText( "artifact" );
         verify( view ).setParentGAV( gavModel.getParent() );
-        verify( view ).disableGroupID( "" );
-        verify( view ).enableArtifactID();
-        verify( view ).disableVersion( "" );
         verify( view ).showParentGAV();
     }
 
@@ -162,6 +165,8 @@ public class POMEditorPanelTest {
         verify( view ).enableArtifactID();
         verify( view ).enableVersion();
         verify( view ).showParentGAV();
-        verify( view ).disableGroupID( anyString() );
-        verify( view ).disableVersion( anyString() );
     }
```

I think that the problem is something related to the way that I'm mocking and that would be a good idea to "resolve" the ticket to "unblock" it. And after that (in fact, right now) I can start investigating why this test has that strange behavior.

**Comment by manstis** [comment]
@karreiro Hi.. it's probably caused by the ```ApplicationPreferences``` test that sets ```CHILD_GAV_EDIT_ENABLED``` to ```true```. Since ```ApplicationPreferences``` uses a ```static``` map the preferences are being retained across all tests. I'd suggest re-initialising the preferences to a "known state" in a ```@Before``` method..

**Comment by manstis** [comment]
Yay, green.. +1 to merge.

**Pull request merged**